Lib/test/test_functools.py

```diff
@@ -480,6 +553,13 @@ class TestPartialCSubclass(TestPartialC):
 class TestPartialPySubclass(TestPartialPy):
     partial = PyPartialSubclass
 
+    def test_subclass_optimization(self):
+        p = py_functools.partial(min, 2)
+        p2 = self.partial(p, 1)
+        assert p2.func is min
```

Comment:
I forgot to say but could you use `self.assert[...]` methods instead of pytest-assert style please?